store/rootmultistore.go

```diff
@@ -440,6 +440,10 @@ func commitStores(version int64, storeMap map[StoreKey]CommitStore) commitInfo {
 		// Commit
 		commitID := store.Commit()
 
+		if commitID.IsZero() {
```

Comment:
Can you explain this line? Is this what the transient store returns? Are we sure this wouldn't skip any "real" commits?